MechJeb2/PartExtensions.cs

```diff
@@ -102,31 +102,17 @@ public static bool IsSepratron(this Part p)
 
         public static bool IsEngine(this Part p)
         {
-            for (int i = 0; i < p.Modules.Count; i++)
-            {
-                PartModule m = p.Modules[i];
-                if (m is ModuleEngines) return true;
-            }
-            return false;
+            return p.FindModuleImplementing<ModuleEngines>() != null;

```

Comment:
Did you test the memory usage of the stock call ? Most of the time when I use loops kile those it is because the stock code allocate memory for each call.
Could you do the same PR without those changes and I ll check the memory uses of the stock call (again)